lib/pxenforcer.js

```diff
@@ -33,28 +34,43 @@ class PxEnforcer {
 
     enforce(req, res, cb) {
         const requestUrl = req.originalUrl;
+        const userAgent = req.get('user-agent');
+        const ipAddress = pxUtil.extractIP(this._config, req);
+
         this.logger.debug('Starting request verification');
 
         if (requestUrl.startsWith(`/${this.reversePrefix}${this._config.FIRST_PARTY_VENDOR_PATH}`)) {
             // reverse proxy client
-            return pxProxy.getClient(req, this._config, pxUtil.extractIP(this._config, req), cb);
+            return pxProxy.getClient(req, this._config, ipAddress, cb);
         }
 
         if (requestUrl.startsWith(`/${this.reversePrefix}${this._config.FIRST_PARTY_XHR_PATH}`)) {
             //reverse proxy xhr
-            return pxProxy.sendXHR(req, this._config, pxUtil.extractIP(this._config, req), this.reversePrefix, cb);
+            return pxProxy.sendXHR(req, this._config, ipAddress, this.reversePrefix, cb);
         }
 
         if (requestUrl.startsWith(`/${this.reversePrefix}${this._config.FIRST_PARTY_CAPTCHA_PATH}`)) {
             // reverse proxy captcha
-            return pxProxy.getCaptcha(req, this._config, pxUtil.extractIP(this._config, req), this.reversePrefix, cb);
+            return pxProxy.getCaptcha(req, this._config, ipAddress, this.reversePrefix, cb);
         }
 
         if (!this._config.ENABLE_MODULE) {
             this.logger.debug('Request will not be verified, module is disabled');
             return cb();
         }
 
+        if (userAgent && this._config.FILTER_BY_USERAGENT.find(item => item.toLowerCase() ===  userAgent.toLowerCase())) {
```

Comment:
in this case you can use `indexOf` or `includes` which are much faster than `find`. just make sure you convert all UAs to lower case when creating the config